tests/test_multipart_encoder.py

```diff
@@ -311,6 +311,12 @@ def test_accepts_custom_headers(self):
         output = m.read().decode('utf-8')
         assert output.index('X-My-Header: my-value\r\n') > 0
 
+    def test_no_parts(self):
+        fields = []
+        boundary = '--90967316f8404798963cce746a4f4ef9'
+        m = MultipartEncoder(fields=fields, boundary=boundary)
+        output = m.read().decode('utf-8')
+        assert output == '----90967316f8404798963cce746a4f4ef9--\r\n'
```

Comment:
I'm not certain that writing anything here is correct behaviour, honestly. I wonder if we should set `self.finished = not self.parts` where `not []` would be `True`.

Reply:
That's good enough for me.